[test plan] Macsec protocol shutdown

parmarkj opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->
This test verifies macsec sessions are up after disabling and enabling macsec feature with macsec already configured.

Summary:
Fixes # (issue)

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

### Approach
#### What is the motivation for this PR?
To add a new test case.

#### How did you do it?
Please refer to the readme file.

#### How did you verify/test it?
Verified on T2.

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?
T2

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

Azarack committed:
Merge pull request #1 from sonic-net/master

Catch up on current master

Azarack committed:
Merge pull request #3 from sonic-net/master

Merge current master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

parmarkj committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

Azarack committed:
Merge branch 'sonic-net:master' into master

parmarkj committed:
macsec protocol shutdown initial commit

BYGX-wcr commented on tests/macsec/test_macsec_protocol_shutdown.py:
```diff
@@ -0,0 +1,108 @@
+    assert len(dut_mka_session) == len(ctrl_links)
+    for port_name, nbr in list(ctrl_links.items()):
+        nbr_mka_session = get_mka_session(nbr["host"])
+        dut_macsec_port = get_macsec_ifname(duthost, port_name)
```
The get_mka_session is only available for virtual SONiC switch. Using it will make this test only applicable for virtual testbed.

BYGX-wcr commented on tests/macsec/test_macsec_protocol_shutdown.py:
```diff
@@ -0,0 +1,108 @@
+    yield setup_info
+
+
+def test_macsec_protocol_shutdown(duthost, macsec_nbrhosts, setup, ctrl_links):
```
You should not directly use duthost feature here and in the definition of setup fixture because this may get you a T2 supervisor when the test is applied to a T2 testbed.

BYGX-wcr commented on tests/macsec/test_macsec_protocol_shutdown.py:
```diff
@@ -0,0 +1,108 @@
+    logger.info("macsec disabled")
+
+    enable_macsec_feature(duthost, macsec_nbrhosts)
+    time.sleep(60)
```
there is no need to sleep and then check macsec container here because the `enable_macsec_feature` function already waits for and checks the launch of macsec containers.

parmarkj committed:
Update test_macsec_protocol_shutdown.py

removed the direct use of duthost feature and removed sleep time post enable_macsec_feature

parmarkj commented:
@BYGX-wcr I have removed the direct use of duthost feature and removed the sleep time post enable_macsec_feature. However, I only have access to virtual testbed, that's why get_macsec_ifname was used. Do you have a replacement recommendation? Thanks
